tests/homestead/identity_precompile/test_identity.py

```diff
@@ -0,0 +1,103 @@
+"""abstract: EIP-2: Homestead Identity Precompile Test Cases."""
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Alloc,
+    Environment,
+    StateTestFiller,
+    Transaction,
+    keccak256,
+)
+from ethereum_test_tools import Opcodes as Op
+
+
+@pytest.mark.with_all_call_opcodes(
+    selector=lambda call_opcode: call_opcode
+    not in [Op.EXTCALL, Op.EXTDELEGATECALL, Op.EXTSTATICCALL]
+    # EXT*CALL opcodes do not contain accept the return data size/offset as an argument
+)
+@pytest.mark.valid_from("Byzantium")
+def test_identity_return_overwrite(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    call_opcode: Op,
+):
+    """Test the return data of the identity precompile overwriting its input."""
+    env = Environment()
+    code = (
+        sum(Op.MSTORE8(offset=i, value=(i + 1)) for i in range(4))  # memory = [1, 2, 3, 4]
```

Comment:
if possible I prefer to have concrete looking memory input like 
`0x000102030405`  because when reading the code we will have to stop and calculate or sometimes we don't see what exactly the result of a calculation and when its not obvious we can have a bug and difficulty to find it.